tests/unit_tests/simpll/DifferentialFunctionComparatorTest.cpp

```diff
@@ -799,56 +747,48 @@ TEST_F(DifferentialFunctionComparatorTest, CmpMemsets) {
-    Function *AuxFL = Function::Create(
-            FunctionType::get(Type::getVoidTy(CtxL),
-                              {Type::getInt32Ty(CtxL), Type::getInt32Ty(CtxL)},
-                              false),
-            GlobalValue::ExternalLinkage,
-            "AuxFL",
-            ModL.get());
-    Function *AuxFR = Function::Create(
-            FunctionType::get(
-                    Type::getVoidTy(CtxR), {Type::getInt32Ty(CtxR)}, false),
-            GlobalValue::ExternalLinkage,
-            "AuxFR",
-            ModR.get());
-
-    BasicBlock *BBL = BasicBlock::Create(CtxL, "", FL);
-    BasicBlock *BBR = BasicBlock::Create(CtxR, "", FR);
-
-    // First compare calls where the additional parameter is not zero.
-    CallInst *CL =
-            CallInst::Create(AuxFL->getFunctionType(),
-                             AuxFL,
-                             {ConstantInt::get(Type::getInt32Ty(CtxL), 5),
-                              ConstantInt::get(Type::getInt32Ty(CtxL), 6)},
-                             "",
-                             BBL);
-    CallInst *CR =
-            CallInst::Create(AuxFR->getFunctionType(),
-                             AuxFR,
-                             {ConstantInt::get(Type::getInt32Ty(CtxR), 5)},
-                             "",
-                             BBR);
-    ASSERT_EQ(DiffComp->testCmpCallsWithExtraArg(CL, CR), 1);
-    ASSERT_EQ(DiffComp->testCmpCallsWithExtraArg(CR, CL), 1);
+TEST_F(DFCLlvmIrTest, CmpCallsWithExtraArgLeft) {
+    auto left = R"(
+        declare i32 @aux(i32, i32) ; called function with extra argument
+        define void @f() {
+            %call1 = call i32 @aux(i32 5, i32 6) ; additional param is not zero
+            %call2 = call i32 @aux(i32 5, i32 0) ; additional param is zero
+            ret void
+        }
+    )";
+    auto right = R"(
+        declare i32 @aux(i32)
+        define void @f() {
+            %call1 = call i32 @aux(i32 5)
+            %call2 = call i32 @aux(i32 5)
+            ret void
+        }
+    )";
+    CREATE_FROM_LLVM(left, right);
+    ASSERT_EQ(testCmpCallsWithExtraArg("call1"), 1);
+    ASSERT_EQ(testCmpCallsWithExtraArg("call2"), 0);
+}
 
-    // Then compare calls when the additional parameter is zero.
-    CL = CallInst::Create(AuxFL->getFunctionType(),
-                          AuxFL,
-                          {ConstantInt::get(Type::getInt32Ty(CtxL), 5),
-                           ConstantInt::get(Type::getInt32Ty(CtxL), 0)},
-                          "",
-                          BBL);
-    CR = CallInst::Create(AuxFR->getFunctionType(),
-                          AuxFR,
-                          {ConstantInt::get(Type::getInt32Ty(CtxR), 5)},
-                          "",
-                          BBR);
-    ASSERT_EQ(DiffComp->testCmpCallsWithExtraArg(CL, CR), 0);
-    ASSERT_EQ(DiffComp->testCmpCallsWithExtraArg(CR, CL), 0);
+TEST_F(DFCLlvmIrTest, CmpCallsWithExtraArgRight) {
```

Comment:
I don't think that we need this one. The functionality is not used much anyways.

Reply:
I am not totally sure how you meant this, remove the `CmpCallsWithExtraArgRight` and leave just `CmpCallsWithExtraArgLeft`? Or did you meant something else?

Reply:
Yeah, just leave the `Left` variant (maybe just name it `CmpCallsWithExtraArg`).